controls/V-71971.rb

```diff
@@ -96,57 +87,53 @@
-  # Make sure semanage is installed
+  describe command('selinuxenabled') do
+    its('exist?') { should be true }
+    its('exit_status') { should eq 0 }
+  end
+
+  describe command('semanage') do
+    its('exist?') { should be true }
+  end
+
+  semanage_results = command('semanage login -l -n')
 
-  describe package("policycoreutils-python") do
-    it { should be_installed }
+  describe semanage_results do
+    its('stdout.lines') { should_not be_empty }
   end
 
-  semanage_results = command("semanage login -l").stdout.split("\n")
-  # Remove Header Row
-  semanage_results.shift
+  semanage_results.stdout.lines.each do |result|
+    login, seuser = result.split(/\s+/)
+
+    # Skip Blank Lines
+    next unless login
 
-  semanage_results.each do |result|
-    result = result.gsub(/\s_/m, ' ').strip.split(" ")
     # Next if for some reason we still have header row
-    next if ( result[0] == 'Login')
+    next if ( login == 'Login')
+
     # Next if root
-    next if ( result[0] == 'root')
-    # Skip Blank Lines
-    next if ( /\S/ !~  result[0])
-
-    if admin_logins.include? "#{result[0]}"
-      describe.one do
-        describe command("semanage login -l | grep #{result[0]}") do
-          its('stdout') { should match %r{sysadm_u} }
-        end
-        describe command("semanage login -l | grep #{result[0]}") do
-          its('stdout') { should match %{system_u} }
-        end
-        describe command("semanage login -l | grep #{result[0]}") do
-          its('stdout') { should match %r{staff_u} }
-        end
-      end
-    elsif non_admin_logins.include? "#{result[0]}"
-      describe.one do
-	      describe command("semanage login -l | grep #{result[0]}") do
-          its('stdout') { should match %r{user_u} }
-        end
-        if( result[0] == '__default__')
-          # all real users should be mapped to a context (i.e. user_u)
-          # but the system isn't forced to map them by default to a context
-          # This will enable defualt to be unconfined like it is by default
-	        describe command("semanage login -l | grep #{result[0]}") do
-            its('stdout') { should match %r{unconfined_u} }
-          end
-        end
-      end
-    # Case when account isn't documented
-    else
-      describe command("semanage login -l | grep #{result[0]}") do
-        its('stdout.strip') { should be_empty }
+    next if ( login == 'root')
+
+    describe "SELinux login #{login}" do
+      # This is required by the STIG
+      if login == '__default__'
+        let(:valid_users){[ 'user_u' ]}
+      elsif admin_logins.include?(login)
+        let(:valid_users){[
+          'sysadm_u',
+          'staff_u'
+        ]}
+      else
+        let(:valid_users){[
+          'user_u',
+          'guest_u',
+          'xguest_u'
+        ]}
       end
+
+      it { expect(seuser).to be_in(valid_users) }
     end
   end
 end
+
+# vim: set expandtab:ts=2:sw=2
```

Comment:
extraneous

Reply:
@samcornwell Sort of. I keep having to retab things and fix indenting and I think that this might help since most of the major editors now support it. @aaronlippold Thoughts?